pkg/controller/pxc/controller.go

```diff
@@ -262,6 +262,7 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(request reconcile.Request) (re
 	}
 
 	haProxySet := statefulset.NewHAProxy(o)
+	haProxyService := pxc.NewServiceHAProxy(o)
```

Comment:
if haproxy is not defined in the cr.yaml it will panic as there is no haproxy.service